analysis/filter/StructureModifier/src/test/kotlin/de/maibornwolff/codecharta/filter/structuremodifier/SubProjectExtractorTest.kt

```diff
@@ -0,0 +1,101 @@
+package de.maibornwolff.codecharta.filter.structuremodifier
+
+import de.maibornwolff.codecharta.model.Project
+import de.maibornwolff.codecharta.serialization.ProjectDeserializer
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.io.File
+
+class SubProjectExtractorTest {
+
+    private lateinit var sampleProject: Project
+
+    @BeforeEach
+    fun serializeProject() {
+        val bufferedReader = File("src/test/resources/sample_project.cc.json").bufferedReader()
+        sampleProject = ProjectDeserializer.deserializeProject(bufferedReader)
+    }
+
+    @Test
+    fun `Non existent path leads to empty project`() {
+        val subProjectExtractor = SubProjectExtractor(sampleProject, null)
+
+        val result = subProjectExtractor.extract("/root/somethig")
+
+        Assertions.assertThat(result.rootNode.children.size).isEqualTo(0)
+    }
+
+    @Test
+    fun `Single path is extracted`() {
+        val subProjectExtractor = SubProjectExtractor(sampleProject, null)
+
+        val result = subProjectExtractor.extract("/root/src/test")
+        println(result)
+
+        val extractedNode = result.rootNode
+        val extractedNodeChild = extractedNode.children.first()
+        Assertions.assertThat(extractedNode.name).isEqualTo("root")
+        Assertions.assertThat(extractedNodeChild.name).isEqualTo("otherFile.java")
+    }
+
+    @Test
+    fun `Attributes of extracted nodes are kept`() {
+        val subProjectExtractor = SubProjectExtractor(sampleProject, null)
+
+        val result = subProjectExtractor.extract("/root/src/test")
+
+        val extractedNodeChild = result.rootNode.children.first()
+        Assertions.assertThat(extractedNodeChild.attributes).containsKey("nloc")
+    }
+
+    @Test
+    fun `Project name is changed if provided`() {
+        val subProjectExtractor = SubProjectExtractor(sampleProject, "foo")
+
+        val result = subProjectExtractor.extract("/root/somethig")
```

Comment:
Typo: "/root/somethig" -> "/root/something"
I know its not important here ;)
again in line 65